File: Data Structures & Algorithms/anagram-groups/submission-9.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        tracker = {}
        res = []
        for s in strs:
            count = [0 for _ in range(26)]
            key = ""
            for c in s:
                count[ord(c) - ord('a')] += 1
            for c in count:
                key += str(c)
                key += "#"
            if key in tracker:
                tracker[key].append(s)
            else:
                tracker[key] = [s]

        for k,v in tracker.items():
            res.append(v)
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-9.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i = 0;
        int j = 1;
        int res = 0;
        while(j < prices.size()){
            if(prices[i] < prices[j]){
                res = max(res, prices[j] - prices[i]);
                j++;
            }
            else{
                i++;
                j++;
            }
        }
        return res;
    }
};
